vividus-extension-selenium/src/main/java/org/vividus/ui/action/search/GenericTextPartFilter.java

```diff
@@ -14,25 +14,31 @@
  * limitations under the License.
  */
 
-package org.vividus.ui.web.action.search;
-
-import javax.inject.Inject;
+package org.vividus.ui.action.search;
 
 import org.openqa.selenium.WebElement;
-import org.vividus.ui.web.action.IWebElementActions;
+import org.vividus.ui.action.ElementActions;
 
-public class TextPartFilter extends AbstractElementFilterAction
+public class GenericTextPartFilter implements IElementFilterAction
 {
-    @Inject private IWebElementActions webElementActions;
+    private final LocatorType locatorType;
+    private final ElementActions elementActions;
 
-    public TextPartFilter()
+    public GenericTextPartFilter(LocatorType locatorType, ElementActions elementActions)
+    {
+        this.locatorType = locatorType;
+        this.elementActions = elementActions;
+    }
+
+    @Override
+    public boolean matches(WebElement element, String textPart)
     {
-        super(WebLocatorType.TEXT_PART);
+        return elementActions.getElementText(element).contains(textPart);
```

Comment:
Do we really need a "contains", could we just use regex for this?

Reply:
I think it can break use cases if someone used in his text part special symbols